### ISSUE TITLE
open-vm-tools not working on Ubuntu 15.10

### ISSUE THREAD

**Issue by superbonaci** [vmware/open-vm-tools#62]
Just installed "sudo apt-get install open-vm-tools" and rebooted ubuntu. Now trying to mount a shared folder but doesn't work:

VMware Workstation
Unable to update run-time folder sharing status:
There was an error mounting the Shared Folders file system inside the guest operating system.

![snip_20160101135741-2](https://cloud.githubusercontent.com/assets/8339596/12070891/f17eccd2-b08f-11e5-8e79-54d6c144f1a0.png)

**Comment by dyno** [comment]
https://bugs.launchpad.net/ubuntu/+source/open-vm-tools/+bug/1500581
it is fixed in open-vm-tools-9.10.3 or 10.0.5 but the fix is not yet picked by ubuntu....

**Comment by dyno** [comment]
```bash
git clone https://github.com/vmware/open-vm-tools.git
cd open-vm-tools
./configure --without-x --without-pam --without-ssl --without-icu
make MODULES=vmhgfs
insmod modules/linux/vmhgfs/vmhgfs.ko

mount -t vmhgfs .host:/ /mnt
ls /mnt
```

**Comment by superbonaci** [comment]
It's NOT fixed yet in debian.

**Comment by dyno** [comment]
i was not saying it was picked up by Debian. to us Debian is just another distro as Ubuntu...

**Comment by superbonaci** [comment]
git clone https://github.com/vmware/open-vm-tools.git
cd open-vm-tools/open-vm-tools
sudo apt-get install libmspack0 libmspack-dev libprocps3 libprocps3-dev dnet-progs libdumbnet-dev doxygen
./configure --without-x --without-pam --without-ssl --without-icu
make MODULES=vmhgfs
insmod modules/linux/vmhgfs/vmhgfs.ko

mount -t vmhgfs .host:/ /mnt
ls /mnt

**Comment by johnkw2275** [comment]
First: THANK YOU superbonaci.  I had the same issue and your post helped me greatly!  Specifically all the previous posts I had seen did not explain how to get insmod to work. Yours worked for me.  I had lost shared folders when I upgraded to 15.10 but it is now back.  However one question.  Every time I reboot I need to do insmod.  It doesn't remember it after reboot.  This is different than what I expected.  Any ideas of what I can check to fix this?

THANK YOU!!!!

**Comment by superbonaci** [comment]
Go to here:

https://github.com/vmware/open-vm-tools/issues/68

**Comment by johnkw2275** [comment]
superbanaci:
THANK YOU!!!!!

**Comment by ivan** [comment]
Updated build instructions for Ubuntu 16.04:

```
sudo apt-get install build-essential libmspack0 libmspack-dev libprocps4-dev dnet-progs libdumbnet-dev doxygen libglib2.0-dev
git clone https://github.com/vmware/open-vm-tools.git
cd open-vm-tools/open-vm-tools
./configure --without-x --without-pam --without-ssl --without-icu --enable-deploypkg=no
make -j4 MODULES=vmhgfs
sudo insmod modules/linux/vmhgfs/vmhgfs.ko
```

**Comment by venkateshshukla** [comment]
Running the command 
> make MODULES=vmhgfs

I get this error

```
root@debian:~/open-vm-tools-master/open-vm-tools# make MODULES=vmhgfs
Making all in lib
make[1]: Entering directory '/root/open-vm-tools-master/open-vm-tools/lib'
Making all in guestRpc
make[2]: Entering directory '/root/open-vm-tools-master/open-vm-tools/lib/guestRpc'
Makefile:331: .deps/nicinfo_xdr.Plo: No such file or directory
make[2]: *** No rule to make target '.deps/nicinfo_xdr.Plo'.  Stop.
make[2]: Leaving directory '/root/open-vm-tools-master/open-vm-tools/lib/guestRpc'
Makefile:330: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/root/open-vm-tools-master/open-vm-tools/lib'
Makefile:361: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by ivan** [comment]
I've seen some really strange build failures (including that one) that can only be resolved by doing a clean checkout of `open-vm-tools` in a new directory.  Not even `git clean -xdf` could fix it.  Really bizarre.

**Comment by harvv** [comment]
TL;DR:  on Ubuntu 14.04 (x64), doing the above seems to build Ok, but no .ko outputs are generated

one minor deviation: running git clone on the current tree as of Feb.9.2017 doesn't result in a "configure" script. I had to follow the snippet from the Readme in the repo, which says:
1. autoreconf -i
2. ./configure
3. Run "make" to build Unix userland and kernel
...though I ran ./configure with the options mentioned here.

it should also be noted that autoreconf/autoconf reports a few warnings of this variety:
libvmtools/Makefile.am:70: warning: source file '$(top_srcdir)/lib/stubs/stub-log.c' is in a subdirectory,
libvmtools/Makefile.am:70: but option 'subdir-objects' is disabled

(and also mentions that the configure.ac appears to be using CFLAGS and not AM_CFLAGS....all of this makes it seem like the maintainer of the open-vm-tools repo maybe fell a bit behind in keeping these things tidy and updated)

Anyone run into the same problem?

**Comment by chartre** [comment]
Same situation than harvv, but in Ubuntu Mate 16.04 (x64) the new problem is that the command:

    $ ./configure --without-x --without-pam --without-ssl --without-icu
    ...
    configure: error: glib2 >= 2.14.0 is required
while:

    $ apt show libglib2.0-0
    Package: libglib2.0-0
    Version: 2.48.2-0ubuntu1
    Priority: standard
    Section: libs
    Source: glib2.0
    Origin: Ubuntu
    ...

**Comment by johnwvmw** [comment]
You probably need to install the libglib2.0-dev package.

Depending what you have installed, you may need one or two additional *-dev packages

**Comment by szachara** [comment]
I will not create new issue for now, I ask here, I get the same error, latest workstation 12.5.6, ubuntu 16.04, open-vm-tools 10.0.7, why this happens?

**Comment by superbonaci** [comment]
The thing is you try to install vmware tools from vmware workstation and it tells you to opt for open-vm-tools. 

https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=2073803

https://blogs.vmware.com/vsphere/2015/09/open-vm-tools-ovt-the-future-of-vmware-tools-for-linux.html

https://communities.vmware.com/thread/547675

**Comment by Brahma108** [comment]
[Solved]

I resolved this issue while using VMware player 14 with Ubuntu 16.0.4.3 LTS by upgrading vmtools. I tried different troubleshooting methods in this article and from elsewhere , downgrading OS and vmware ,etc, but with no luck, the total, and a bit strange process below::

1. installed vmware player 14 at first install, shared folders did not work. Originally vmware installed VMtools version 10.0.7.52125, which works at home on VMware workstation 12.5 but did not work here, huh.

2. Reinstalled vmware player 14 after performing other methods, when opening vmware guest in player a popup to upgrade vmtools came about i downloaded and installed tools version 10.1.15 , then reinstalled on the existing Ubuntu OS having issues. NOTE:: When reinstalling Do not do the default install process in OS as you will need to overwrite the existing hgfs and hgfsclient configurations [yes]. just pay attention to the install process wherever hgfs is mentioned overwrite the configuration

3. Reboot. then ah HA! all is good and shared folders works without issue.
